drt_1d.py

```diff
@@ -32,47 +34,152 @@ def construct_source_term(num_voxels:int, source_voxel_index:int, source_strengt
+    |   ri         |                   |         rj  |
+        |__________|                   |__________|
+         left distance                  right distance
+         
+                  
+      <--  Streaming to the left <--
+    ----------------------------------------------------
+    | End Voxel    | In-between voxels | Start Voxel | 
+    ----------------------------------------------------
+    |---[]---------|-------------------|---------[]--|
+    |   rj         |                   |         ri  |
+        |__________|                   |__________|
+         left distance                  right distance
     """
-    logging.debug("Calculating K_trans from voxel %d to voxel %d with voxel size %f", ri, rj, voxel_size)
+    logging.debug("Calculating K_trans from index %d to index %d, position %.3f to position %.3f, with voxel size %f", start_index, end_index, ri, rj, voxel_size)
     
     if voxel_size <= 0:
         raise ValueError("Voxel size must be positive and non-zero.")
-    if ri < 0 or ri >= len(sigma_s) or rj < 0 or rj >= len(sigma_s):
-        raise ValueError("Voxel indices must be non-negative and within the range of the sigma_s_matrix.")
+    if ri < 0 or ri > len(sigma_s) * voxel_size or rj < 0 or rj > len(sigma_s) * voxel_size:
+        raise ValueError("Positions must be within the spatial domain defined by the voxels.")
     
-    min_index = min(ri, rj)
-    max_index = max(ri, rj)
-    distance = abs(ri - rj) * voxel_size
-    tau = np.sum(sigma_s[min_index:max_index]) * voxel_size
-    streaming_operator = np.exp(-tau) / (4 * np.pi * distance**2)
+    distance = abs(ri - rj)
+    if distance == 0:
+        raise ValueError(f"The positions of (ri,start_index):({ri},{start_index}) and (rj,end_index):({rj},{end_index}) are the same value and will result in an infinite streaming operator.")
+    tau = 0
+    
+    if start_index == end_index:
+        # Same voxel calculation
+        tau = sigma_s[start_index] * distance
+        logging.debug("Intra-voxel tau: %f", tau)
+    else:
+        # Multiple voxel calculation
+        if ri < rj:
+            # Direction is to the right (start_index and ri are on the left)
+            
+            # Calculate left fraction
+            if ri == start_index*voxel_size:
+                # ri is on the left-most point of the voxel
+                left_distance = voxel_size
+            elif start_index*voxel_size < ri and ri < (start_index+1)*voxel_size:
+                # ri is between the voxel edges
+                left_distance = (start_index+1)*voxel_size - ri
+            elif ri == (start_index+1)*voxel_size:
+                # ri is on the right-most point of the voxel
+                left_distance = 0
+            else:
+                # ri is not within or on the edges of this voxel
+                raise ValueError(f"Invalid position (ri: {ri}) and voxel index (start_index: {start_index}) combination.")
```

Comment:
Why isn't this always
```
left_distance = (start_index + 1) * voxel_size - ri
```
no need for all these conditionals, and you'll never have a `ValueError` if you computer `start_index` from `ri`